lib/matplotlib/axis.py

```diff
@@ -1223,18 +1221,29 @@ def get_ticklines(self, minor=False):
         return self.get_majorticklines()
 
     def get_majorticklocs(self):
-        "Get the major tick locations in data coordinates as a numpy array"
+        """Get the array of major tick locations in data coordinates."""
         return self.major.locator()
 
     def get_minorticklocs(self):
-        "Get the minor tick locations in data coordinates as a numpy array"
-        return self.minor.locator()
+        """Get the array of minor tick locations in data coordinates."""
+        # Remove minor ticks duplicating major ticks.
+        major_locs = self.major.locator()
```

Comment:
why not pass the `major_locs` here, if we don't want to do it in the locator?  Agree that its an API change, but I suspect a minor one; this method is not used elsewhere, except in the tests...